ironic/drivers/modules/drac/raid.py

```diff
@@ -1064,14 +1066,35 @@ def _create_virtual_disks(task, node):
     return _commit_to_controllers(node, controllers)
 
 
-def controller_supports_ehba_mode(settings, controller_fqdd):
-    # check the raid controller mode
-    raid_controller_mode = settings.get(
-        "{}:RAIDCurrentControllerMode".format(controller_fqdd))
-    raid_controller_mode = raid_controller_mode.current_value
+def _controller_in_hba_mode(raid_settings, controller_fqdd):
+    controller_mode = raid_settings.get(
+        '{}:{}'.format(controller_fqdd, _CURRENT_RAID_CONTROLLER_MODE))
+
+    if "Enhanced HBA" in controller_mode.current_value:
+        return True
+    else:
+        return False
+
+
+def _controller_supports_ehba_mode(settings, controller_fqdd):
+    # check the raid controller suppports HBA mode
+    raid_controller_mode = {sett: value.possible_values for sett, value in
+                            settings.items() if _CURRENT_RAID_CONTROLLER_MODE
+                            in sett}
 
-    # if e-HBA found then return True
-    if "Enhanced HBA" in raid_controller_mode:
+    if not raid_controller_mode:
+        # Raid controller does not support HBA mode
+        return False
+
+    raid_cntrl_attr = "{}:{}".format(
+        controller_fqdd,_CURRENT_RAID_CONTROLLER_MODE)
+
+    raid_cntrl_attr = raid_cntrl_attr if raid_cntrl_attr\
+        in raid_controller_mode.keys() else None
+
+    if  raid_cntrl_attr and "Enhanced HBA" in raid_controller_mode[
+        raid_cntrl_attr]:
+        # raid controller supports HBA mode
```

Comment:
This seems overly complicated.  Couldn't this all be reduced to:
raid_cntrl_attr = "{}:{}".format(controller_fqdd,_CURRENT_RAID_CONTROLLER_MODE)
current_cntrl_mode = settings.get(raid_cntrl_attr)
if not current_cntrl_mode:
    return False
elif "Enhanced HBA" in current_cntrl_mode.possible_values:
   return True
else:
    return False

Reply:
Yes. Done Thanks